StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1208SystemUsingDirectivesMustBePlacedBeforeOtherUsingDirectives.cs

```diff
@@ -50,93 +53,78 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+                if (string.Equals(SystemUsingDirectiveName, firstIdentifierInUsingDirective?.Text, StringComparison.Ordinal))
                 {
-                    continue;
+                    if (systemUsingDirectivesShouldBeBeforeThisName != null)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptor, usingDirective.GetLocation(), GetNamespaceNameWithoutAlias(usingDirective.Name.ToString()), systemUsingDirectivesShouldBeBeforeThisName));
+                        continue;
+                    }
+
+                    var previousUsing = usings[i - 1];
+                    var firstIdentifierInPreviousUsingDirective = GetFirstIdentifierInUsingDirective(previousUsing);
+
+                    if (!string.Equals(SystemUsingDirectiveName, firstIdentifierInPreviousUsingDirective?.Text, StringComparison.Ordinal) || previousUsing.StaticKeyword.Kind() != SyntaxKind.None)
+                    {
+                        systemUsingDirectivesShouldBeBeforeThisName = GetNamespaceNameWithoutAlias(previousUsing.Name.ToString());
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptor, usingDirective.GetLocation(), GetNamespaceNameWithoutAlias(usingDirective.Name.ToString()), systemUsingDirectivesShouldBeBeforeThisName));
+                    }
                 }
-
-                string @namespace = namespaceSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
-                string precedingNamespace = precedingNamespaceSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
-
-                // Using directive for '{namespace}' must appear before directive for '{precedingNamespace}'
-                context.ReportDiagnostic(Diagnostic.Create(Descriptor, syntax.GetLocation(), @namespace, precedingNamespace));
-                break;
             }
         }
 
-        private static string GetTopLevelNamespace(SemanticModel semanticModel, UsingDirectiveSyntax syntax, out INamespaceSymbol namespaceSymbol, CancellationToken cancellationToken)
-        {
-            SymbolInfo symbolInfo = semanticModel.GetSymbolInfo(syntax.Name, cancellationToken);
-            namespaceSymbol = symbolInfo.Symbol as INamespaceSymbol;
-            if (namespaceSymbol == null)
-            {
-                return null;
-            }
+        private static bool ExcludeGlobalKeyword(IdentifierNameSyntax token) => !token.Identifier.IsKind(SyntaxKind.GlobalKeyword);
 
-            string fullyQualifiedName = namespaceSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-            string name = fullyQualifiedName;
+        private static SyntaxToken? GetFirstIdentifierInUsingDirective(UsingDirectiveSyntax usingDirective)
+        {
+            return usingDirective.DescendantNodes().OfType<IdentifierNameSyntax>().FirstOrDefault(ExcludeGlobalKeyword)?.Identifier;
+        }
 
+        private static string GetNamespaceNameWithoutAlias(string name)
```

Comment:
In PR #786 @vweijsters implemented a helper method that does exactly that. It might be a good idea to wait until his PR is merged. You could use that to reduce code dublication